robot/src/main/java/frc/robot/OI.java

```diff
@@ -7,36 +7,70 @@
 
 package frc.robot;
 
+import edu.wpi.first.wpilibj.XboxController;
+import edu.wpi.first.wpilibj.GenericHID.Hand;
+
 /**
  * This class is the glue that binds the controls on the physical operator
  * interface to the commands and command groups that allow control of the robot.
  */
 public class OI {
-  //// CREATING BUTTONS
-  // One type of button is a joystick button which is any button on a
-  //// joystick.
-  // You create one by telling it which joystick it's on and which button
-  // number it is.
-  // Joystick stick = new Joystick(port);
-  // Button button = new JoystickButton(stick, buttonNumber);
-
-  // There are a few additional built in buttons you can use. Additionally,
-  // by subclassing Button you can create custom triggers and bind those to
-  // commands the same as any other Button.
-
-  //// TRIGGERING COMMANDS WITH BUTTONS
-  // Once you have a button, it's trivial to bind it to a button in one of
-  // three ways:
-
-  // Start the command when the button is pressed and let it run the command
-  // until it is finished as determined by it's isFinished method.
-  // button.whenPressed(new ExampleCommand());
-
-  // Run the command while the button is being held down and interrupt it once
-  // the button is released.
-  // button.whileHeld(new ExampleCommand());
-
-  // Start the command when the button is released and let it run the command
-  // until it is finished as determined by it's isFinished method.
-  // button.whenReleased(new ExampleCommand());
+    // Assuming we use an XBOX controller (the logitechs should map similarly)
+    // The drive controller should be the very first one listed in the DriverStation
+    private XboxController mDriveController = new XboxController(0);
+
+    /**
+     * Get requested X-axis movement speed from the controller
+     * @return requested speed
+     */
+    public double getDriveSpeed() {
+        // left stick, Y axis
+        return mDriveController.getRawAxis(2); 
```

Comment:
Let's use constants here. If we subclass as suggested we could use `mDriveController.getRawAxis(Constants.XBoxController. LEFT_X_AXIS)` which, while longer, is much clearer as to which controller value is being read.